BepressImportPlugin.inc.php

```diff
@@ -110,54 +121,54 @@ function executeCLI($scriptName, &$args) {
 			if (!is_dir($volumePath) || preg_match('/^\./', $volumeName) || !$volumeName) continue;
 
 			// Import issues from oldest to newest
-			$importIssues = array();
+			$importIssues = [];
 			$issueHandle = opendir($volumePath);
 			while ($importIssues[] = readdir($issueHandle));
 			sort($importIssues, SORT_NATURAL);
 			closedir($issueHandle);
 
-			$allIssueIds = array();
+			$allIssueIds = [];
 			$curIssueId = 0;
 
 			foreach ($importIssues as $issueName){
 				$issuePath = $volumePath . '/' . $issueName;
 				if (!is_dir($issuePath) || preg_match('/^\./', $issueName) || !$issueName) continue;
 
 				// Import articles from oldest to newest
-				$importArticles = array();
+				$importArticles = [];
 				$articleHandle = opendir($issuePath);
 				while ($importArticles[] = readdir($articleHandle));
 				sort($importArticles, SORT_NATURAL);
 				closedir($articleHandle);
 
 				$currSectionId = 0;
-				$allSectionIds = array();
+				$allSectionIds = [];
 
 				foreach ($importArticles as $entry) {
 					$articlePath = $issuePath . '/' . $entry;
-					if (!is_dir($articlePath) || preg_match('/^\./', $entry)) continue;
+					if (!is_dir($articlePath) || preg_match('/^\./', $entry) || !$entry) continue;
```

Comment:
I think it would be better to fix the loop with an extra variable (e.g. `for ($importArticles = []; $path = readdir($articleHandle); $importArticles[] = $path);`), but that's just my preference 😁

Reply:
The `preg_match('/^\./', $entry)` could ignore a good folder (e.g. `.xxx`), but if I remember well this plugin expects folders to have numbers, right? Then perhaps it's better to check if the entry is a valid number, otherwise emit a warning.

Reply:
The `preg_match('/^\./', $entry)` could ignore a good folder (e.g. `.xxx`), but if I remember well this plugin expects folders to be numbered, right? Then perhaps it's better to check if the entry is a valid number, otherwise emit a warning.

Reply:
> I think it would be better to fix the loop with an extra variable (e.g. `for ($importArticles = []; $path = readdir($articleHandle); $importArticles[] = $path);`), but that's just my preference 😁

@jonasraoni, just curious: what would that look like in this case? That's not a pattern I've typically used.

Reply:
I thought you've added the `!$entry` to cover the case of [`readdir()` returning false in the last iteration](https://github.com/mfelczak/bepress/pull/29/files#diff-7d725ffa810b550abf276122996b130e1b7b2ccbe8f119efe6c42ad3c08d2f3cR140), if that's not the purpose, then ignore my comment =]

> what would that look like in this case?

I've left a piece of untested code in the comment to fix the loop, that should probably be applied to the other places.
In fact, here I would replace the `readdir()` by the [`FilesystemIterator`](https://www.php.net/manual/en/class.filesystemiterator.php), which will skip the dot folders automatically \o/